src/ObjLoading/ObjContainer/SoundBank/SoundBankWriter.cpp

```diff
@@ -0,0 +1,285 @@
+#include "SoundBankWriter.h"
+
+#include "Crypto.h"
+#include "ObjContainer/SoundBank/SoundBankTypes.h"
+#include "Sound/WavTypes.h"
+#include "Utils/Alignment.h"
+#include "Utils/FileUtils.h"
+
+#include <cstring>
+#include <filesystem>
+#include <iostream>
+#include <unordered_map>
+
+std::unordered_map<unsigned int, unsigned char> INDEX_FOR_FRAMERATE{
+    {8000,   0},
+    {12000,  1},
+    {16000,  2},
+    {24000,  3},
+    {32000,  4},
+    {44100,  5},
+    {48000,  6},
+    {96000,  7},
+    {192000, 8},
+};
+
+class SoundBankWriterImpl : public SoundBankWriter
+{
+    static constexpr char BRANDING[] = "Created with OAT - OpenAssetTools";
+    static constexpr int64_t DATA_OFFSET = 0x800;
+    static constexpr uint32_t MAGIC = FileUtils::MakeMagic32('2', 'U', 'X', '#');
+    static constexpr uint32_t VERSION = 14u;
+
+    inline static const std::string PAD_DATA = std::string(16, '\x00');
+
+public:
+    explicit SoundBankWriterImpl(const std::string& fileName, std::ostream& stream, ISearchPath* assetSearchPath)
+        : m_file_name(fileName),
+          m_stream(stream),
+          m_asset_search_path(assetSearchPath),
+          m_sounds(),
+          m_current_offset(0),
+          m_total_size(0),
+          m_entry_section_offset(0),
+          m_checksum_section_offset(0)
+    {
+    }
+
+    void AddSound(const std::string& soundFilePath, unsigned int soundId, bool looping, bool streamed) override
+    {
+        this->m_sounds.push_back({soundFilePath, soundId, looping, streamed});
```

Comment:
```cpp
this->m_sounds.emplace_back(soundFilePath, soundId, looping, streamed);
```

Reply:
Changing to this breaks the build

Reply:
Ahh okay you made this a struct without an explicit constructor. I usually try to avoid that and use classes with constructors. That makes sure that there cant be some odd behaviour with std::string not having its constructor called and makes sure that the members have default values set.

If you do this:
```cpp
class SoundBankEntryInfo
{
public:
	SoundBankEntryInfo()
		: m_sound_id(0u),
		  m_looping(false),
		  m_streamed(false)
	{
	}

	SoundBankEntryInfo(std::string filePath, const unsigned int soundId, const bool looping, const bool streamed)
		: m_file_path(std::move(filePath)),
		  m_sound_id(soundId),
		  m_looping(looping),
		  m_streamed(streamed)
	{
	}

	std::string m_file_path;
	unsigned int m_sound_id;
	bool m_looping;
	bool m_streamed;
};
```
That suggestion will work, avoids a copy of the path string (i think it currently does one, not 100% on that one though) and is more in-line with how the rest of the codebase works